app/src/main/java/com/aconno/sensorics/ui/settings/virtualscanningsources/MqttVirtualScanningSourceActivity.kt

```diff
@@ -240,13 +243,26 @@ class MqttVirtualScanningSourceActivity : BaseActivity() {
         val id = if (mqttVirtualScanningSourceModel == null) 0 else mqttVirtualScanningSourceModel!!.id
         val enabled = mqttVirtualScanningSourceModel?.enabled != false
 
+        val portAsInteger = try {
+            port.toInt().also {
+                if(it < 0 || it > MAX_PORT_NUMBER) throw NumberFormatException("Port invalid")
```

Comment:
There is no need for try/catch here, as you could use if statement. Also NumberFormatException is not a correct exception to be thrown in this situation - everything is fine with the number format, but the number is out of range.

You should put the validation in port's EditTextView like Roberto suggested. Then you don't need this part of the code.

Reply:
port is a String. It was possible to make it very long so port.toInt() throws a NumberFormatException.
Now I restricted EditTextView to 5 digits and removed the try-catch.

Reply:
You should not show a toast, but you should put a validation on EditText.

https://medium.com/@bendaniel10/mini-project-android-edittext-validation-9bd8ab04576a

Reply:
Good idea! Changed it to edittext.error